### ISSUE TITLE
Remove legacy data services and functionality

### ISSUE THREAD

**Issue by tmadlener** [key4hep/k4FWCore#107]
#100 Introduced several Legacy components and services that should be absorbed cleanly into the new Frame based I/O. Once all of that has been done, the legacy functionality can be removed.

**Comment by hegner** [comment]
@andresailer - anything against doing the cleanup now? We have the podio PR ready to kick out the EventStore

**Comment by tmadlener** [comment]
Tagging @kjvbrt and @BrieucF here as well as I am not entirely sure what the status is for the repositories in HEP-FCC.

**Comment by andresailer** [comment]
Can you summarize what the status would be after the cleanup? Can old files still be read, or only new files?

**Comment by tmadlener** [comment]
IIRC only new files will be readable. There is #103 still open which at least started to address this. There are two things to keep in mind.
1. Adding an automatic switch to a `podio::ROOTLegacyReader` is probably not to hard to implement
2. That might not be enough because the `PodioOutput`/`PodioLegacyDataSvc` combination wrote a slightly different file format that the `EventStore`/`podio::ROOTWriter` combination did. This is the underlying issue in #95 AFAIK.

Final consideration: Solving #105 might change things once more (at least on the python config side), so maybe we can try to make a few of the breaking changes in one go and then fix up things.

**Comment by kjvbrt** [comment]
As far as I can tell we should be ready with the repositories under HEP-FCC (all converted to use Frame based DataService). And for large scale samples -- they are read with FCCAnalysis, so they don’t depend on k4FWCore.

One case where legacy reader in k4FWCore is needed is, that someone has local simulated samples with k4SimGeant4 and need to be able to run reconstruction.

**Comment by tmadlener** [comment]
Fixed by #171
